src/Testing/CurlRunnings.hs

```diff
@@ -81,32 +86,35 @@ runCase state@(CurlRunningsState _ _ _ tlsCheckType) curlCase = do
       eInterpolatedQueryParams = interpolateViaJSON state $ fromMaybe (QueryParameters []) (queryParameters curlCase)
   case (eInterpolatedUrl, eInterpolatedHeaders, eInterpolatedQueryParams) of
     (Left err, _, _) ->
-      return $ CaseFail curlCase Nothing Nothing [QueryFailure curlCase err]
+      return $ CaseFail curlCase Nothing Nothing [QueryFailure curlCase err] 0
     (_, Left err, _) ->
-      return $ CaseFail curlCase Nothing Nothing [QueryFailure curlCase err]
+      return $ CaseFail curlCase Nothing Nothing [QueryFailure curlCase err] 0
     (_, _, Left err) ->
-      return $ CaseFail curlCase Nothing Nothing [QueryFailure curlCase err]
+      return $ CaseFail curlCase Nothing Nothing [QueryFailure curlCase err] 0
     (Right interpolatedUrl, Right interpolatedHeaders, Right (QueryParameters interpolatedQueryParams)) ->
       case sequence $ runReplacements state <$> requestData curlCase of
         Left l ->
-          return $ CaseFail curlCase Nothing Nothing [QueryFailure curlCase l]
+          return $ CaseFail curlCase Nothing Nothing [QueryFailure curlCase l] 0
         Right replacedJSON -> do
           initReq <- parseRequest $ T.unpack interpolatedUrl
           manager <- newManager noVerifyTlsManagerSettings
 
-          let request =
-                setRequestBodyJSON (fromMaybe emptyObject replacedJSON) .
-                setRequestHeaders (toHTTPHeaders interpolatedHeaders) .
-                appendQueryParameters interpolatedQueryParams  .
-                (if tlsCheckType == DoTLSCheck then id else (setRequestManager manager)) $
-                initReq {method = B8S.pack . show $ requestMethod curlCase}
+          let !request =
```

Comment:
For my own learning - could you explain why the `BangPattern` is required or desirable here (I've never used this Haskell feature before)?

Reply:
Good question! I'm actually not yet entirely sure that it is indeed needed, still experimenting with this PR a bit and should've marked it as such. 

The idea is that in the code that is timing the request:

```
          start <- nowMillis
          response <- httpBS request
          stop <- nowMillis
          let elapsed = stop - start
```

If `request` is lazy, then part of this timer will be spent evaluating the thunk that `request` is pointing to. I _think_ that by making it strict, it will be evaluated upon declaration, and the timing should be more accurate. I still need to play around with the code a little more to make sure this is correct. I know that the `seq` function can also be used to enforce strictness, but still figuring this all out myself too :)

Reply:
Thanks for the explanation 🥇